blockindex/contractstaking/bucket_info.go

```diff
@@ -39,6 +39,18 @@ func (bi *bucketInfo) Deserialize(b []byte) error {
 	return bi.loadProto(&m)
 }
 
+// clone clones the bucket info
+func (bi *bucketInfo) clone() *bucketInfo {
+	return &bucketInfo{
+		TypeIndex:  bi.TypeIndex,
+		CreatedAt:  bi.CreatedAt,
+		UnlockedAt: bi.UnlockedAt,
+		UnstakedAt: bi.UnstakedAt,
+		Delegate:   bi.Delegate,
+		Owner:      bi.Owner,
```

Comment:
`owner, delegate` are interface/pointer, will need to duplicate it as well

otherwise `owner` will be contaminated in https://github.com/iotexproject/iotex-core/blob/master/blockindex/contractstaking/event_handler.go#L441-L443 ?

Reply:
fixed